app/models/ddr/batch/batch_object_role.rb

```diff
@@ -0,0 +1,36 @@
+module Ddr::Batch
+
+  class BatchObjectRole < ActiveRecord::Base
+    belongs_to :batch_object, :inverse_of => :batch_object_roles
+
+    OPERATION_ADD = "ADD".freeze  # Add the principal and role to the object in the indicated scope
+    OPERATIONS = [ OPERATION_ADD ].freeze
+
+    validates :operation, inclusion: { in: OPERATIONS }
+    with_options if: :operation_requires_agent? do |obj|
+      obj.validates :agent, presence: true
+    end
+    with_options if: :operation_requires_role_type? do |obj|
+      obj.validates :role_type, inclusion: { in: Ddr::Auth::Roles.role_types.map(&:title) }
+    end
+    with_options if: :operation_requires_role_scope? do |obj|
+      obj.validates :role_scope, inclusion: { in: Ddr::Auth::Roles::SCOPES }
+    end
+    validate :policy_role_scope_only_for_collections, if: "role_scope == Ddr::Auth::Roles::POLICY_SCOPE"
+
+    def operation_requires_agent?
```

Comment:
Why is this and similar methods necessary?  If the operation requires a role, then the role must be valid, and the validity of a role is defined in the Role class.